homeassistant/components/home_connect/switch.py

```diff
@@ -161,3 +167,64 @@ async def async_update(self) -> None:
         else:
             self._state = None
         _LOGGER.debug("Updated, new state: %s", self._state)
+
+
+class HomeConnectSettingSwitch(HomeConnectEntity, SwitchEntity):
+    """Setting switch class for Home Connect."""
+
+    def __init__(self, device, setting, desc, icon):
+        """Initialize the entity."""
+        super().__init__(device, desc)
+        self._state = None
+        self._setting = setting
+        self._icon = icon
```

Comment:
In generally these should be converted to the more modern _attr_** options. And the getters removed. Applies to the number platform too.